src/main/kotlin/org/rust/lang/core/psi/ext/RsDocAndAttributeOwner.kt

```diff
@@ -79,24 +83,120 @@ inline val RsDocAndAttributeOwner.attributeStub: RsAttributeOwnerStub?
  * Returns [QueryAttributes] for given PSI element.
  */
 val RsDocAndAttributeOwner.queryAttributes: QueryAttributes
+    get() = getQueryAttributes(null)
+
+/** [explicitCrate] is passed to avoid resolve triggering */
+fun RsDocAndAttributeOwner.getQueryAttributes(
+    explicitCrate: Crate?,
+    stub: RsAttributeOwnerStub? = attributeStub
+): QueryAttributes {
+    testAssert { !DumbService.isDumb(project) }
+    return when {
+        stub?.hasAttrs == false -> QueryAttributes.EMPTY
+        stub?.hasCfgAttr == false -> rawAttributesUnchecked
+        else -> getQueryAttributesNoStub(explicitCrate)
```

Comment:
Maybe add a comment on why we should distinguish between these three cases? I mean, to notice that only in case of `cfg_attr`, we really have to evaluate something